asserts/snapasserts/validation_sets.go

```diff
@@ -63,8 +63,9 @@ func (e *ValidationSetsConflictError) Error() string {
 // ValidationSetsValidationError describes an error arising
 // from validation of snaps against ValidationSets.
 type ValidationSetsValidationError struct {
-	// MissingSnaps maps missing snap names to the validation sets requiring them.
-	MissingSnaps map[string][]string
+	// MissingSnaps maps missing snap names to the expected revisions and respective validation sets requiring them.
+	// Revisions may be unset if no specific revision is required
+	MissingSnaps map[string]map[snap.Revision][]string
```

Comment:
if that assumption is right, is still the case that we could have a situation like:

snap-a: {
     3:  {val-set-1, val-set-2}
     Unset:  {val-set-3}
}

we don't have  a test like that though?

Reply:
Yes, I will add one. Thanks

Reply:
Sorry, it's actually covered by TestCheckInstalledSnapsErrorFormat.

Reply:
Ah, got you, indeed, I thought the concern was about requiring a specific revision & unset revision at the same time.

Will add.

Reply:
Added.